packages/data-controls/src/index.js

```diff
@@ -91,19 +107,134 @@ export function dispatch( ...args ) {
  * import * as resolvers from './resolvers';
  *
  * registerStore( 'my-custom-store', {
- * 	reducer,
- * 	controls,
- * 	actions,
- * 	selectors,
- * 	resolvers,
+ * reducer,
+ * controls,
+ * actions,
+ * selectors,
+ * resolvers,
  * } );
  * ```
- *
  * @return {Object} An object for registering the default controls with the
- *                  store.
+ * store.
  */
 export const controls = {
+	ACQUIRE_STORE_LOCK: ( { scope, exclusive } ) => {
+		const promise = new Promise( ( resolve ) => {
+			lockRequests.unshift( {
+				exclusive,
+				scope,
+				notifyAcquired: resolve,
+			} );
+		} );
+		acquireAvailableLocks();
+		return promise;
+	},
+
+	RELEASE_STORE_LOCK: ( { lock } ) => {
+		releaseLock( lock );
+		// Run anything that got unblocked by releasing this lock in the next tick:
+		setTimeout( () => {
```

Comment:
Why do we have to wait a tick?

Reply:
So that the current operation may be completed before we start the next one. That's not required, but makes the flow of things easier to reason about.